App views, Navbar and Headers

amrllkmn opened the pull request:
The pull request will add:
- App views:
    - Data ( for the historical data to be displayed )
    - Home ( indoor air quality will be rendered here)
    - Settings ( for users to customise the application )
- Navbar ( navigating between the views )
- Header ( the app's logo will be rendered here )

The PR closes issues #38, #39, #40.

amrllkmn committed:
app/views: Create home screen

amrllkmn committed:
app/views: Add Data screen

The Data screen will render the charts that display the historical data.

amrllkmn committed:
app/views: Add Settings screen

The settings screen will allow users to customise and configure the app.

amrllkmn committed:
app/views: Add bottom navigation bar

The bottom navigation bar will allow the users to navigate to the
different views of the app.

amrllkmn committed:
app/views: Moved navbar from App.js

The NavBar is separated from the App.js into a separate file to keep the
code readable

amrllkmn committed:
app/views: Add icons to navbar

The navbar now has icons which changes from an outline to a filled icon
whether it is active or otherwise.

amrllkmn committed:
app/views: Fix formatting

amrllkmn committed:
app/views: Ignore workspace preferences

amrllkmn committed:
app/views: Rename folder

amrllkmn committed:
app/tests: Fix test packages

Changing the preset for Jest to work with Expo

amrllkmn committed:
app/tests: Add navbar test

amrllkmn committed:
app/tests: Reorganise test folders

The test folders have been moved to its respective view or component for
a more organised folder structure

amrllkmn committed:
app/views: Add header component

The header is passed as a prop to the screen components. The
documentation requires the views to be wrapped by the Stack component
for the header to be visible.

amrllkmn committed:
app/views: Add height to header style

amrllkmn committed:
app/tests: Fix test suites

Due to the components having to be wrapped with a Stack component, it
caused the tests to fail. This commit fixes the issue

amrllkmn committed:
app/views: Refactor header

Refactoring the header to make it easier for testing. It would also make
it easier for a global style to be applied in the future.

amrllkmn committed:
app/tests: Add header test

roryort reviewed (commented):
Only thing I would say is that the disclaimer at the top of each file needs to be implemented properly. I didn't see the disclaimer in each file and where I did see it it mentioned that it was for the monitoring server. If you can add the disclaimer for each file and ensure that you mention that it's for the App then I'll be good to approve. Other than that everything is good. (Sorry I know I am nit-picking here)

Sevenarth commented on mobile-app/src/views/Settings.js:
```diff
@@ -20,12 +20,17 @@ import { StatusBar } from "expo-status-bar";
+                headerStyle: styles.headerStyle,
+            }}
+        >
             <SettingsStack.Screen name="Home" component={Settings} />
```
Settings not Home

amrllkmn replied:
I'm resolving this conversation and re-requesting a review.

Sevenarth reviewed (changes requested):
Perhaps you could add a brief documentation for _nearly every_ function (?)

Globalise the stylesheets into a single file, so that there are not duplicates.

amrllkmn committed:
app/views: Add documentation

amrllkmn committed:
app/views: Globalise StyleSheet object

Since the components have similar styles throughout the app, separating it into a central style makes it easier to implement changes.

amrllkmn commented:
Made the requested changes.
